.github/workflows/plugins-ci.yml

```diff
@@ -93,13 +98,189 @@ jobs:
 
   test:
     name: Test
+    if: >
+      !failure() &&
+      !cancelled() &&
+      inputs.database == 'none'
     runs-on: ${{ matrix.os }}
     permissions:
       contents: read
     strategy:
       matrix:
         node-version: [14, 16, 18]
         os: [macos-latest, ubuntu-latest, windows-latest]
+
+    steps:
+      - name: Check out repo
+        uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+
+      - name: Setup Node ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - name: Install dependencies
+        run: npm i --ignore-scripts
+
+      - name: Run tests
+        run: npm test
+
+  test-postgres:
+    if: >
+      !failure() &&
+      !cancelled() &&
+      inputs.database == 'postgres'
```

Comment:
I must say that I'm not a fan of this approach because it will show many skipped steps

I know that the dynamic step is not straightforward, but I saw this impl:
https://github.com/pkgjs/action

Reply:
that makes it more complex. On the other hand I dont think, that it will show the skipped steps, because we already skip on jobs-levels. Anyhow. I will close this and just create the corresponding redis and mongo versions.